api/endpoints.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 The go-pttai Authors
+// This file is part of the go-pttai library.
+//
+// The go-pttai library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-pttai library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-pttai library. If not, see <http://www.gnu.org/licenses/>.
+
+package api
+
+import (
+    "net"
+    "net/http"
+
+    "github.com/ethereum/go-ethereum/rpc"
+)
+
+// StartHTTPEndpoint starts the HTTP RPC endpoint, configured with cors/vhosts/modules
+func StartHTTPEndpoint(endpoint string, apis []rpc.API, modules []string, cors []string, vhosts []string, timeouts rpc.HTTPTimeouts) (net.Listener, *rpc.Server, *http.Server, error) {
```

Comment:
If this is only for JSON RPC, maybe rename the file to `rpc/endpoints.go`?

Reply:
renamed api as rpc.